msprime/simulations.py

```diff
@@ -963,6 +964,69 @@ def mean_recombination_rate(self):
 
         return np.average(rates, weights=weights)
 
+    def slice(self, start=None, end=None, remove_flanks=False):
+        """
+        Returns a subset of this recombination map between the specified end
+        points. If start is None, it defaults to 0. If end is None, it defaults
+        to the end of the map. If remove_flanks is True, remove the flanking
+        zero recombination rate regions such that the sequence length of the
+        new recombination map is end - start.
+        """
+        positions = self.get_positions()
+        rates = self.get_rates()
+
+        if start is None:
+            i = 0
+            start = 0
+        if end is None:
+            end = positions[-1]
+            j = len(positions)
+
+        # TODO: Is there interest in supporting negative indexes? Eg. make
```

Comment:
I guess we could right? In the case where the recombination map is discrete it makes perfect sense, and I guess the same logic should hold for floating point values. Here's the standard approach:
```
    def __getitem__(self, index):
        if index < 0:
            index += len(self)
        if index < 0 or index >= len(self):
            raise IndexError("Index out of bounds")
        return self.getter(index)
```